File: windows/Models/Transaction.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace FinanceAccountant.Models;

public class Transaction
{
    [Key]
    public int Id { get; set; }

    public TransactionType Type { get; set; }

    public string Category { get; set; } = string.Empty;

    public decimal Amount { get; set; }

    public DateTime Date { get; set; }

    public string? Note { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.Now;
}

public enum TransactionType
{
    Income,  // 收入
    Expense  // 支出
}

File: android/app/src/main/java/com/finance/accountant/domain/model/Transaction.kt
package com.finance.accountant.domain.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "transactions")
data class Transaction(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val type: TransactionType,
    val category: String,
    val amount: Double,
    val date: Long,  // timestamp
    val note: String? = null,
    val createdAt: Long = System.currentTimeMillis()
)

enum class TransactionType {
    INCOME,
    EXPENSE
}